drivers/syslog/syslog_write.c

```diff
@@ -94,227 +98,118 @@ static bool syslog_safe_to_block(void)
-                            {
-                              return ret;
-                            }
-
-                          ret = channel->sc_ops->sc_write_force(channel,
-                                                                "\r\n", 2);
-                          if (ret < 0)
-                            {
-                              return ret;
-                            }
-
-                          nwritten = head + 1;
-                        }
+                      continue;
                     }
-#endif
 
-                  if (nwritten < buflen)
+                  ret = write(channel, buffer + nwritten, head - nwritten);
+                  if (ret >= 0)
                     {
-                      ssize_t ret;
-
-                      ret = channel->sc_ops->sc_write_force(channel,
-                                                          buffer + nwritten,
-                                                          buflen - nwritten);
-                      if (ret < 0)
-                        {
-                          return ret;
-                        }
-                      else
-                        {
-                          nwritten += ret;
-                        }
+                      ret = write(channel, "\r\n", 2);
                     }
-                }
-              else
-                {
-                  DEBUGASSERT(channel->sc_ops->sc_force != NULL);
-                  for (nwritten = 0; nwritten < buflen; nwritten++)
-                    {
-#ifdef CONFIG_SYSLOG_CRLF
-                      /* Check for LF */
 
-                      if (buffer[nwritten] == '\n' &&
-                          !(channel->sc_state & SYSLOG_CHANNEL_DISABLE_CRLF))
-                        {
-                          /* Add CR */
+                  if (ret < 0)
+                    {
+                      return ret;
+                    }
 
-                          channel->sc_ops->sc_force(channel, '\r');
-                        }
+                  nwritten = head + 1;
+                }
+            }
 #endif
 
-                      channel->sc_ops->sc_force(channel, buffer[nwritten]);
-                    }
+          if (nwritten < buflen)
+            {
+              ret = write(channel, buffer + nwritten, buflen - nwritten);
+              if (ret < 0)
+                {
+                  return ret;
+                }
+              else
+                {
+                  nwritten += ret;
                 }
             }
         }
-    }
-  else
-    {
-      int i;
-
-      for (i = 0; i < CONFIG_SYSLOG_MAX_CHANNELS; i++)
+      else
         {
-          FAR syslog_channel_t *channel = g_syslog_channel[i];
-          nwritten = 0;
-
-          if (channel == NULL)
-            {
-              break;
-            }
-
-#ifdef CONFIG_SYSLOG_IOCTL
-          if (channel->sc_state & SYSLOG_CHANNEL_DISABLE)
+          putc = !force ? channel->sc_ops->sc_putc :
```

Comment:
```suggestion
          putc = !force ? channel->sc_ops->sc_force :
                          channel->sc_ops->sc_putc;
```

Reply:
if branch is fast than else, most of the time we call sc_putc()